### PR TITLE
pivotal-173792650: Replace base path for multiple absolute paths 

### PR DESCRIPTION
**Pull request by Juan-EBI**
https://www.pivotaltracker.com/story/show/173792650

### CODE REVIEW

File: commons/commons-bio/src/main/kotlin/ebi/ac/uk/paths/SubmissionFolderResolver.kt [hunk 1]
```diff
@@ -1,25 +1,22 @@
...
+    private val submissionFolder: Path,
+    private val ftpFolder: Path
+) {
 
-    fun getSubmissionFtpFolder(submission: ExtSubmission): Path =
-        basePath.resolve(FTP_FOLDER).resolve(submission.relPath)
-
-    fun getSubmissionFolder(submission: ExtSubmission): Path =
-        basePath.resolve(SUBMISSION_PATH).resolve(submission.relPath)
+    fun getSubmissionFtpFolder(submissionRelPath: String): Path =
```
Comment (jhoanmanuelms) [inline commons/commons-bio/src/main/kotlin/ebi/ac/uk/paths/SubmissionFolderResolver.kt:12]: Silly detail but, don't these two methods fit in a single line?

File: submission/submission-webapp/src/itest/resources/application.yml [hunk 2]
```diff
@@ -18,7 +18,9 @@ spring:
     port: 5672
 
 app:
-  basepath:
+  submissionPath:
+  ftpPath:
+  magicPath:
```
Comment (jhoanmanuelms) [inline submission/submission-webapp/src/itest/resources/application.yml:23]: Don't we need this property in the application's main config file?

Also, I didn't see it's being used in the code, only in the `ApplicationProperties` class but I assume it's the user magic folder? In any case, I think the property name can be improved or at least add a comment indicating what it's used for.

File: submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/common/config/GeneralConfig.kt [hunk 3]
```diff
@@ -10,6 +10,7 @@ import org.springframework.context.annotation.Configuration
 @Configuration
 @EnableConfigurationProperties(ApplicationProperties::class)
 internal class GeneralConfig {
+
```
Comment (jhoanmanuelms) [inline submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/common/config/GeneralConfig.kt:13]: image

File: submission/submission-webapp/src/itest/resources/application.yml (hunk 2 above)
Comment (Juan-EBI) [reply submission/submission-webapp/src/itest/resources/application.yml:23]: ohh yes let me remove this one, is for future changes

File: submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/common/config/GeneralConfig.kt (hunk 3 above)
Comment (Juan-EBI) [reply submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/common/config/GeneralConfig.kt:13]: hahaha

**Pull request merged**